build-support/bin/prune_travis_cache.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash -e
+
+function nuke_if_too_big() {
+  path=$1
+  limit_mb=$2
+  echo "## Pruning ${path} (if larger than ${limit_mb}MB)..."
+  if [[ -d "${path}" ]]; then
+    actual_mb=$(du -m -d0 "${path}" | cut -f 1)
+    echo "Size of ${path}: ${actual_mb}MB"
+    if (( actual_mb > limit_mb )); then
+      echo "${path} is too large, nuking it."
+      rm -rf "${path}"
+    else
+      echo "${path} is not too large, leaving it."
+    fi
+  else
+    echo "Directory ${path} doesn't exist."
+  fi
+}
+
+
+# Prune the dirs we cache on travis if they get too big.
+# Note that some of these dirs are only relevant to native code building shards, and others only to
+# pants running shards. And some only to linux or only to osx.
+# However there's no harm in checking them all in all cases.
+
+nuke_if_too_big "${HOME}/.pants_pyenv" 512
+nuke_if_too_big "${HOME}/.aws_cli" 128
```

Comment:
We can safely rename this to `"${AWS_CLI_ROOT}"`. `.travis.yml` will have already populated the env var.